### PR TITLE
APITestClass creation

### PR DESCRIPTION
**Pull request by plvsl**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        <maven.compiler.source>15</maven.compiler.source>
+        <maven.compiler.target>15</maven.compiler.target>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>7.4.0</version>
+
```
Comment (Orkyboy) [inline pom.xml]: empty line

please reformat code

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>com.googlecode.json-simple</groupId>
+            <artifactId>json-simple</artifactId>
+            <version>1.1.1</version>
+        </dependency>
+
+    </dependencies>
+</project>
```
Comment (Orkyboy) [inline pom.xml]: add empty line at the end of file

File: src/test/java/APITestClass.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.*;
+
+
+public class APITestClass {
+
+    @BeforeTest
```
Comment (Orkyboy) [inline src/test/java/APITestClass.java]: why not `@BeforeClass`

File: src/test/java/APITestClass.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+                .get("/{id}", "a")
+                .then()
+                .assertThat()
+                .statusCode(400)
+                .and()
+                .body("errors.id[0]", equalTo("The value 'a' is not valid."));
+    }
+
+    @Test
+    public void checkPostRequestWorkCorrectly() {
```
Comment (Orkyboy) [inline src/test/java/APITestClass.java]: so I suggest adding various combination of fields to test 
like empty userName, empty password, empty id, id greater than the biggest integer

File: src/test/java/APITestClass.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+                .body(requestBody.toJSONString())
+                .when()
+                .post()
+                .then()
+                .assertThat()
+                .statusCode(201);
+    }
+
+    @Test
+    public void checkPutRequestWorkCorrectly() {
```
Comment (Orkyboy) [inline src/test/java/APITestClass.java]: one more test case - add the entity which already exists

File: src/test/java/APITestClass.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+                .when()
+                .get( "/{id}")
+                .then()
+                .assertThat()
+                .body("userName", equalTo("Polina"))
+                .body("password", equalTo("Vasileva"));
+    }
+
+    @Test
+    public void checkDeleteRequestWorkCorrectly() {
```
Comment (Orkyboy) [inline src/test/java/APITestClass.java]: what if you pass delete without id? Will it delete everything?

**Review by Orkyboy — APPROVED** [review]

File: src/test/java/APITestClass.java (hunk 6 above)
Comment (plvsl) [reply src/test/java/APITestClass.java]: Made the test to check it throws an error

File: README.md [hunk 7]
```diff
@@ -1 +1,45 @@
...
+
+## Task
+
+Write and submit a small, runnable suite of integration tests for the web service endpoints mentioned below:
+ 
+GET : https://fakerestapi.azurewebsites.net/api/v1/Users
+GET : https://fakerestapi.azurewebsites.net/api/v1/Users/{id}
+POST : https://fakerestapi.azurewebsites.net/api/v1/Users
+
+Request body schema
```
Comment (Orkyboy) [inline README.md:11]: nit: you can use readme formatting to convert jsons to code blocks and links to more readable format

File: README.md [hunk 8]
```diff
@@ -1 +1,45 @@
...
+
+### Instructions to run tests 
+
+- Open a terminal or command prompt
+- Navigate to **Project Root**
+- Execute `mvn clean test`
+
+## Bugs
+
+- POST request returns invalid Status Code (200 instead of 201)
```
Comment (Orkyboy) [inline README.md]: Please leave more details about bugs, it shows how you would create them for prod systems

so 
Bug title
Description
Steps
Expected
Actual
etc.

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.2</version>
+                <configuration>
+                    <source>15</source>
+                    <target>15</target>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (Orkyboy) [inline pom.xml]: empty line is not resolved!

File: src/test/java/APITestSuit/UsersDeleteTests.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+import org.testng.annotations.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.is;
+
+public class UsersDeleteTests extends BaseTest {
+
+    @Test
+    public void checkDeleteResponseWithValidId() {
+
```
Comment (Orkyboy) [inline src/test/java/APITestSuit/UsersDeleteTests.java]: empty line

File: src/test/java/APITestSuit/BaseTest.java [hunk 11]
```diff
@@ -0,0 +1,18 @@
+package APITestSuit;
+
+import io.restassured.RestAssured;
+import org.testng.annotations.BeforeClass;
+
+public class BaseTest {
+
+    protected String userName = "userName";
```
Comment (Orkyboy) [inline src/test/java/APITestSuit/BaseTest.java]: can be static final

File: src/test/java/APITestSuit/UsersPutTests.java [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+                .put(idURL, "1")
+                .then()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .and()
+                .body(id, equalTo(1))
+                .body(userName, equalTo("Polina"))
+                .body(password, equalTo("PolinaPassword"));
+
+        // check database result
```
Comment (Orkyboy) [inline src/test/java/APITestSuit/UsersPutTests.java]: I'd change it to `check data state`

File: src/test/java/APITestSuit/UsersPutTests.java [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+                .when()
+                .get(idURL, "1")
+                .then()
+                .assertThat()
+                .body(id, equalTo(1))
+                .body(userName, equalTo("Polina"))
+                .body(password, equalTo("PolinaPassword"));
+    }
+
+}
```
Comment (Orkyboy) [inline src/test/java/APITestSuit/UsersPutTests.java]: empty line add please

**Review by Orkyboy — APPROVED** [review]

File: README.md [hunk 14]
```diff
@@ -1 +1,45 @@
...
+  "password": "string"
+}
+
+DELETE: https://fakerestapi.azurewebsites.net/api/v1/Users/{id}
+
+## Instructions
+
+### Instructions to run tests 
+
+- Open a terminal or command prompt
```
Comment (Orkyboy) [inline README.md:45]: I'd add `prerequisites` section also which says what user should have pre installed (maven, java?)

**Pull request merged**